src/arrays.jl

```diff
@@ -23,16 +23,16 @@ function deepdiff(X::Vector, Y::Vector)
     # substrings.
 
     lengths = zeros(Int, length(X)+1, length(Y)+1)
-    backtracks = fill(:nothing, axes(lengths))
-    backtracks[1,2:end] .= :X
-    backtracks[2:end,1] .= :Y
-    backtracks[1,1] = :nothing
+    backtracks = fill((0, 0), axes(lengths))
```

Comment:
Doesn't help readability. A enum or a custom struct will probably be better at getting storage efficiency and readibility

Reply:
The choice of representation here is that it corresponds with index offsets here: https://github.com/ssfrr/DeepDiffs.jl/pull/15/files#diff-b13f4801e308864f0bd76da7a4928cc9R55

I believe any other choice would involve an eventual conversation to `(0/1, 0/1)`, because that's really what they represent in the algorithm: walk backwards in X, Y, or both, or neither. Which isn't to say other choices wouldn't be better. Did you have any enum ideas?

I see the storage efficiency concern as separate (and still important). But I think that is best addressed by storing `Tuple{Bool, Bool}` instead of `Tuple{Int, Int}`.

`sizeof(Tuple{Bool, Bool}) == 2`. You could pack the bits more efficiently, but then I would expect time efficiency to suffer.

Reply:
Yeah, seems like this would pretty much work as-is as `fill((false,false), axes(lengths))`.

Reply:
probably want to add a comment though that explains what the two fields represent

Reply:
Yeah, seems like this would pretty much work as-is as `fill((false,false), axes(lengths))`.